### PR TITLE
Add Todoist v8.0.1

### PR DESCRIPTION
**Pull request by nihaals**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

Todoist was removed and rejected due to it being MAS only (#20024) but is now available as a .dmg again.

### CODE REVIEW

File: Casks/todoist.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cask "todoist" do
+  version "8.0.1"
+  sha256 "1cd6db17e5610f1af5de2898e56a47733141b553838cd1ed6f9acbd59668bca8"
+
+  url "https://electron-dl.todoist.com/mac/Todoist-#{version}.dmg"
+  name "Todoist"
+  desc "To-do list"
+  homepage "https://todoist.com/home"
```
Comment (nihaals) [inline Casks/todoist.rb:8]: https://todoist.com/home shows the home page even if you're logged in, while https://todoist.com takes you to the app if you're logged in

File: Casks/todoist.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+cask "todoist" do
+  version "8.0.1"
+  sha256 "1cd6db17e5610f1af5de2898e56a47733141b553838cd1ed6f9acbd59668bca8"
+
+  url "https://electron-dl.todoist.com/mac/Todoist-#{version}.dmg"
+  name "Todoist"
+  desc "To-do list"
```
Comment (nihaals) [inline Casks/todoist.rb:7]: The description I was most unsure on

File: Casks/todoist.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  homepage "https://todoist.com/home"
+
```
Comment (miccal) [inline Casks/todoist.rb:9]: Please add the following `livecheck` block:
```suggestion

  livecheck do
    url "https://electron-dl.todoist.com/mac/latest-mac.yml"
    strategy :electron_builder
  end

```

**Review by miccal — CHANGES_REQUESTED** [review]

**Comment by miccal** [comment]
Thank you for your contribution to `homebrew-cask` @nihaals.

Could you please try to add a [`zap` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap), thanks.

**Comment by nihaals** [comment]
I've added a `zap` using [homebrew-createzap](https://github.com/nrlquaker/homebrew-createzap). It looks longer than normal, possibly due to the included Safari extension?

Is the commit message for b35615af20428e9a34f47087c052f6eb9695fceb fine or does it also need to include the cask name?

File: Casks/todoist.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  zap trash: [
+    "~/Library/Application Scripts/S3DD273774.com.todoist.mac-beta.Todoist.extensions",
+    "~/Library/Application Scripts/S3DD273774.com.todoist.mac-dev.Todoist.extensions",
+    "~/Library/Application Scripts/S3DD273774.com.todoist.mac.Todoist",
+    "~/Library/Application Scripts/S3DD273774.com.todoist.mac.Todoist.extensions",
+    "~/Library/Application Scripts/com.todoist.mac.Todoist",
+    "~/Library/Application Scripts/com.todoist.mac.Todoist.TodoistSafari",
+    "~/Library/Application Scripts/com.todoist.mac.Todoist.TodoistShare",
+    "~/Library/Application Scripts/com.todoist.mac.Todoist.TodoistToday",
```
Comment (miccal) [inline Casks/todoist.rb:25]:
```suggestion
    "~/Library/Application Scripts/*com.todoist.mac*",
```

File: Casks/todoist.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    "~/Library/Application Support/Todoist",
+    "~/Library/Containers/com.todoist.mac.Todoist",
+    "~/Library/Containers/com.todoist.mac.Todoist.TodoistSafari",
+    "~/Library/Containers/com.todoist.mac.Todoist.TodoistShare",
+    "~/Library/Containers/com.todoist.mac.Todoist.TodoistToday",
```
Comment (miccal) [inline Casks/todoist.rb:30]:
```suggestion
    "~/Library/Containers/com.todoist.mac.Todoist*",
```

File: Casks/todoist.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    "~/Library/Group Containers/S3DD273774.com.todoist.mac-beta.Todoist.extensions",
+    "~/Library/Group Containers/S3DD273774.com.todoist.mac-dev.Todoist.extensions",
+    "~/Library/Group Containers/S3DD273774.com.todoist.mac.Todoist",
+    "~/Library/Group Containers/S3DD273774.com.todoist.mac.Todoist.extensions",
```
Comment (miccal) [inline Casks/todoist.rb:34]:
```suggestion
    "~/Library/Group Containers/*com.todoist.mac*",
```

**Review by miccal — CHANGES_REQUESTED** [review]

**Comment by miccal** [comment]
We can clean up the `zap` stanza with wildcards :)

**Comment by nihaals** [comment]
That feels so obvious now 😔

**Comment by miccal** [comment]
Thank you @nihaals.

**Pull request merged**